lib/grape/middleware/base.rb

```diff
@@ -21,8 +21,39 @@ def call(env)
       def call!(env)
         @env = env
         before
+
         @app_response = @app.call(@env)
-        after || @app_response
+
+        # Merge Headers from API into @header
+        unless @header.nil?
```

Comment:
Should be able to refactor this into methods. These similarly manipulate `@app_response`.